org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.xtend

```diff
@@ -219,6 +222,14 @@ class RenameService2 implements IRenameService2 {
 		return null
 	}
 
+	protected def String getConvertedValue(EObject grammarElement, ILeafNode leaf) {
+		switch (grammarElement) {
+			RuleCall: valueConverterService.toValue(leaf.text, grammarElement.rule.name, leaf).toString()
+			CrossReference: getConvertedValue(grammarElement.terminal, leaf)
```

Comment:
shouldt we consult `linkingHelper.getRuleNameFrom(updatable.getCrossReference())` e.g. as in referenceupdater.getQualifiedName

Reply:
Maybe even use `org.eclipse.xtext.linking.impl.LinkingHelper.getCrossRefNodeAsString(INode, boolean)` and find a proper cross ref node starting at the leaf node.

Reply:
As you wish.